executions/graphql-kotlin-automatic-persisted-queries/src/test/kotlin/com/expediagroup/graphql/apq/preparsed/ApolloPersistedQuerySupportAsyncTest.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.apq.preparsed
+
+import com.expediagroup.graphql.apq.fixture.ProductGraphQL
+import graphql.ExecutionInput
+import graphql.execution.preparsed.persisted.PersistedQueryNotFound
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class ApolloPersistedQuerySupportAsyncTest {
+    @Test
+    fun `Using ApolloPersistedQuerySupportAsync should return error when no query with provided hash is in the cache`() {
+
+        // First execution fails to find persisted query string
+
+        val extensions = mapOf(
+            "persistedQuery" to mapOf(
```

Comment:
Should this string and data model be saved in source code so we can reference it?

```kt
const val APQ_EXTENSION_KEY = "persistedQuery"

data class APQExtension(val version: Int, val sha256Hash: String)
```